packages/test-app-frontend/src/App/ITwinJsApp/ITwinJsApp.tsx

```diff
@@ -96,7 +96,17 @@ function useIModel(
       IModelApp.authorizationClient = authorizationClient;
 
       let disposed = false;
-      const iModelPromise = CheckpointConnection.openRemote(iTwinId, iModelId);
+      const iModelPromise = iModelId === ""
```

Comment:
I like this. Create a promise on a ternary and store it in a variable for later use. :D Feel free to resolve.